Server/src/models/DatasetModels/DatasetDeleteModel.ts

```diff
@@ -0,0 +1,276 @@
+            await this.deleteMaterialsQuery(materialsToDelete)
+            await this.deleteCompositions(rawCompositionIds)
+        }
+    }
+
+    private async deleteCompositions(rawCompositionIds: any) {
+        let isCompositionInUse: any
+        let compositionsToDelete: number[] = []
+        for (let index = 0; index < rawCompositionIds.length; index++) {
+            isCompositionInUse = await this.selectOneUseOfCompositionQuery(rawCompositionIds.id)
+            if (isCompositionInUse == undefined) {
+                compositionsToDelete.push(rawCompositionIds[index].id)
+            }
+        }
+        if (compositionsToDelete.length > 0) {
+            await this.deleteCompositionsQuery(compositionsToDelete)
+        }
+    }
+
+    async deleteDatasetDataType(datasetDataTypeId: number) {
+        let isDatasetDataTypeInUse = await this.selectOneUseOfDatasetDataTypeQuery(datasetDataTypeId)
+        if (isDatasetDataTypeInUse == undefined) {
+            await this.deleteDatasetDataTypeQuery(datasetDataTypeId)
+        }
+    }
+
+    async deleteAuthorsOfPublication(publicationsId: number) {
+        let isAuthorInUse: any
+        let authorsToDelete: number[] = []
+        let rawAuthorIds = await this.connection.query("SELECT authorsId FROM publications_authors_authors WHERE publicationsId = ?", [publicationsId])
+        await this.connection.query("DELETE FROM publications_authors_authors WHERE publicationsId = ?", [publicationsId])
+        // Check which authors are still in use
+        for (let index = 0; index < rawAuthorIds.length; index++) {
+            isAuthorInUse = await this.connection.query("SELECT publicationsId FROM publications_authors_authors WHERE authorsId = ?", [rawAuthorIds[index].authorsId]);
+            if (isAuthorInUse.length < 1) {
+                authorsToDelete.push(rawAuthorIds[index].authorsId)
+            }
+        }
+        if (authorsToDelete.length > 0)
+            await this.deleteAuthorsQuery(authorsToDelete)
+    }
+
+    async deletePublication(publicationId: number) {
+        let isPublicationInUse = await this.selectOneUseOfPublicationQuery(publicationId)
+        if (isPublicationInUse == undefined) {
+            await this.deletePublicationQuery(publicationId)
+        }
+    }
+    async deletePublisher(publisherId: number) {
+        let isPublisherInUse = await this.selectOneUseOfPublisherQuery(publisherId)
+        if (isPublisherInUse == undefined) {
+            await this.deletePublisherQuery(publisherId)
+        }
+    }
+
+    async deletePublicationType(publicationTypeId: number) {
+        let isPublicationTypeInUse = await this.selectOneUseOfPublicationTypeQuery(publicationTypeId)
+        if (isPublicationTypeInUse == undefined) {
+            await this.deletePublicationTypeQuery(publicationTypeId)
+        }
+    }
+
+    async rejectDataset(id: number): Promise<string> {

```

Comment:
hopefully this wont cause issues, seems like it could be a source of bugs if not properly done, will need some testing for sure to assure this deletes a dataset properly without remains or without ruining other datasets, but im sure you guys did it well :P

Reply:
It carries out the "order" of deletion in the same manner as how our test migration does. It definitely needs a lot of testing but it falls back on the concept of the data set as the core of everything. 
I wish you were at the group PR review as it was explained in depth during it, in part to address these specific concerns.